addon/components/hyper-table-v2/manage-columns.ts

```diff
@@ -0,0 +1,91 @@
+import { A } from '@ember/array';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import TableHandler from '@upfluence/hypertable/core/handler';
+import { ColumnDefinition } from '@upfluence/hypertable/core/interfaces';
+
+type ManageColumn = {
+  definition: ColumnDefinition;
+  visible: boolean;
+};
+
+interface HyperTableV2ManageColumnsArgs {
+  handler: TableHandler;
+}
+
+export default class HyperTableV2ManageColumns extends Component<HyperTableV2ManageColumnsArgs> {
+  @tracked displayAvailableFields: boolean = false;
+  @tracked _activeColumnCategory: null | string = null;
+  @tracked _searchColumnDefinitionKeyword: null | string = null;
+
+  get _columnCategories(): string[] {
+    return this.args.handler.columnDefinitions
+      .reduce((categories: string[], columnDefinition) => {
+        if (!categories.includes(columnDefinition.category)) {
+          categories.push(columnDefinition.category);
+        }
+        return categories;
+      }, [])
+      .sort((a, b) => a.localeCompare(b));
+  }
+
+  get _orderedFilteredClusters() {
+    let fields = A(
+      this.args.handler.columnDefinitions.filter((columnDefinition) => {
+        const search = (this._searchColumnDefinitionKeyword || '').toLowerCase();
+        const hasSearched =
+          !this._searchColumnDefinitionKeyword || columnDefinition.name.toLowerCase().indexOf(search) >= 0;
+        const hasActiveGroup = !this._activeColumnCategory || columnDefinition.category === this._activeColumnCategory;
+        return hasActiveGroup && hasSearched;
+      })
+    ).sortBy('name');
+
+    return this.groupByClusteringKey(fields);
+  }
+
+  groupByClusteringKey(columnDefinitions: ColumnDefinition[]) {
+    const map = new Map();
+
+    columnDefinitions.forEach((columnDefinition) => {
+      const cluster = map.get(columnDefinition.clustering_key);
+      const manageColumn: ManageColumn = {
+        definition: columnDefinition,
+        visible: !!this.args.handler.columns.find((column) => column?.definition.key === columnDefinition.key)
```

Comment:
The `!!` notation could be misleading for some people.
I suggest using the isEmpty util from ember instead - it will make this more readable.